ibc-clients/ics07-tendermint/src/client_state/execution.rs

```diff
@@ -344,3 +367,61 @@ where
 
     Ok(())
 }
+
+/// Update the `client_state`'s ID, trusting period, latest height, processed height,
+/// and processed time metadata values to those values provided by a verified substitute
+/// client state in response to a successful client recovery.
+///
+/// Note that unlike the `update_on_upgrade` function, `update_on_recovery` assumes
+/// that the client being updated has already been re-initialised such that its original
+/// client and consensus states have been overwritten to their new states.
+///
+/// This function is typically implemented as part of the [`ClientStateExecution`]
+/// trait, but has been made standalone in order to enable greater flexibility
+/// of the ClientState APIs.
+pub fn update_on_recovery<E>(
+    subject_client_state: ClientStateType,
+    ctx: &mut E,
+    subject_client_id: &ClientId,
+    substitute_client_state: Any,
+) -> Result<(), ClientError>
+where
+    E: TmExecutionContext,
+    E::ClientStateRef: From<ClientStateType>,
+    // E::ClientStateRef: TryInto<ClientStateType, Error = ClientError>,
+    E::ConsensusStateRef: ConsensusStateConverter,
+{
+    let substitute_client_state = ClientStateType::try_from(substitute_client_state)?;
```

Comment:
Can we use `ClientState` to be consistent with `update_on_upgrade`?

Reply:
[81f9833](https://github.com/cosmos/ibc-rs/pull/1127/commits/81f9833876d671ec6da31771aad8b1904225febf)